lib/thredded/email_notifier.rb

```diff
@@ -0,0 +1,53 @@
+module Thredded
+  class EmailNotifier
+
+    def new_post(post, users)
+      @post = post
+      users_opted_in = exclude_those_opting_out_of_followed_activity_notifications(users)
+      return if users_opted_in.empty?
+      PostMailer.post_notification(post.id, users_opted_in.map(&:email)).deliver_now
+      MembersMarkedNotified.new(post, users_opted_in).run
+    end
+
+    def new_private_post(post, users)
+      @post = post
+      @topic = post.postable
+      users = private_topic_recipients
```

Comment:
You're passing in users, so no need for @topic at all, instead, just inline the two lines you need from private_topic_recipients (lines 26, 27)